content/chainguard/fips/not-approved-algorithms.md

```diff
@@ -0,0 +1,170 @@
+
+In all of the above usecases digest calculation does not provide any security functionality, it is meant to detect accidental corruption or provide speed ups. Overall, data is typically protected by SHA2-256 and is transmitted over a secure and authenticated TLS channel.
+
+Note that using CRC32C or MD5 doesn't even performance gains, as specialist functions exist for non-security purposes with significantly higher performace, such as [XXHASH](https://xxhash.com/). In most cases, non-security functionality should upgrade from CRC32C, MD5, SHA1 to XXHASH3.
+
+However, if one must have interoperability with existing formats and services, and it is established that digest usage is for non-security purposes, one must be able to access the insecure digests. Chainguard is integrating support for such use cases for MD5 and SHA1 across our FIPS images. Each language, application implementation is very different and specific, and it is documented below.
+
+Although SHA1 is currently approved, it is already deprecated by RFCs and NIST is deprecating SHA1 by 2030, thus below implementations are forward looking and attempt address access to MD5 today and SHA1 in the future.
+
+SHA1 is available as approved in Chainguard FIPS Provider for OpenSSL versions 3.0.9, 3.1.2 and 3.4.0. It is not-approved starting version 3.6.0. The below guidance will apply to SHA1 as well likely in year 2027.
+
+## Access to MD5 for non-security purposes
+
+### OpenSSL FIPS and MD5
+
+In Chainguard FIPS images, OpenSSL operates in approved only mode and has default property query string set to `fips=yes` for all algorithms and services, such as message digests, HMAC, and so on.
+
+MD5 is available as not-approved service. One can request access to it on opt-in basis using `?fips=yes` (preffer fips implementation if there is one, and fallback to a non-fips one) or `-fips` (disregard request for a fips implementation, and return any available one) property query strings via C-API or via command line options to calculate message digest. Usage of these digests in higher level algorithms are blocked, for example whilst `openssl dgst` calculation is allowed, `openssl dsgst -sign | -verify | -hmac` are blocked.
+
+Examples:
+
+```shell
+$ echo chainguard | openssl dgst -md5
+Error setting digest
+801B3D417B7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:375:Global default library context, Algorithm (MD5 : 89), Properties ()
+801B3D417B7F0000:error:03000086:digital envelope routines:evp_md_init_internal:initialization error:crypto/evp/digest.c:271:
+```
+
+```shell
+$ printf chainguard | openssl dgst -propquery -fips -md5
+MD5(stdin)= f8d689ee8221617c032b0e71f9c597ac
+```
+
+Note that calculating HMAC is still blocked:
+
+```shell
+$ printf chainguard | openssl dgst -propquery -fips -md5 -hmac averylonghmackey
```

Comment:
The examples should all `echo` or `printf` so the digests are the same. Or `echo -n` I suppose.